src/tools/run-make-support/src/diff/mod.rs

```diff
@@ -58,15 +66,24 @@ impl Diff {
         self
     }
 
+    pub fn normalize(&mut self, regex: &str, replacement: &str) -> &mut Self {
+        self.normalizers.push((regex.to_owned(), replacement.to_owned()));
+        self
+    }
+
     /// Executes the diff process, prints any differences to the standard error.
     #[track_caller]
     pub fn run(&self) {
         let expected = self.expected.as_ref().expect("expected text not set");
-        let actual = self.actual.as_ref().expect("actual text not set");
+        let mut actual = self.actual.as_ref().expect("actual text not set").to_string();
         let expected_name = self.expected_name.as_ref().unwrap();
         let actual_name = self.actual_name.as_ref().unwrap();
+        for (regex, replacement) in &self.normalizers {
+            let re = Regex::new(regex).expect("bad regex in custom normalization rule");
+            actual = re.replace_all(&actual, replacement).into_owned();
```

Comment:
should we assert that we actually replace something?

Reply:
As you prefer.